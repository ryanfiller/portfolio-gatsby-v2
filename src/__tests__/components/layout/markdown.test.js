import React from 'react'
import Markdown from '../../../components/layout/markdown'
import Image from '../../../components/markdown/image'

import mockPost from '../../../__mocks__/post-mock'

describe('<Markdown />', () => {
  const props = () => ({
    post: mockPost,
    className: 'test-class'
  })

  it('should render correctly', async () => {
    const component = mount(<Markdown {...props()} />) 
    expect(component.find('article.test-class').length).toEqual(1)
    expect(component.html()).toMatchSnapshot()
    expect(await axe(component.html())).toHaveNoViolations()
  })

  it('should render React components correctly', async () => {
    const component = mount(<Markdown {...props()} />) 
    expect(component.find(Image).length).toEqual(8)
  })

})