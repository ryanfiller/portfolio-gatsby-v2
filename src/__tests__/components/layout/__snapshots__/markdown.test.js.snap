// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown /> should render correctly 1`] = `
"<article class=\\"test-class\\"><h1>Heading level 1</h1><h2>Heading level 2</h2><h3>Heading level 3</h3><h4>Heading level 4</h4><h5>Heading level 5</h5><h6>Heading level 6</h6><p>Nulla vitae elit libero, a pharetra augue. Maecenas sed diam eget risus varius blandit sit amet non magna. Maecenas sed diam eget risus varius blandit sit amet non magna. Aenean lacinia bibendum nulla sed consectetur.</p><p>Cras mattis consectetur purus sit amet fermentum. Vestibulum id ligula porta felis euismod semper. Maecenas sed diam eget risus varius blandit sit amet non magna. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Cras mattis consectetur purus sit amet fermentum. Praesent commodo cursus magna, vel scelerisque nisl consectetur et.</p><p><em>Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</em>
<em>Sed posuere consectetur est at lobortis.</em>
<em><em>Nulla vitae elit libero, a pharetra augue.</em></em></p><blockquote><p>Donec ullamcorper nulla non metus auctor fringilla. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum.</p></blockquote><ul><li>Donec sed odio dui.</li><li>Morbi leo risus, porta ac consectetur ac, vestibulum at eros.</li><li>Lorem ipsum dolor sit amet, consectetur adipiscing elit.<ul><li>Donec sed odio dui.</li><li>Morbi leo risus, porta ac consectetur ac, vestibulum at eros.</li><li>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</li></ul></li></ul><ol><li>Donec sed odio dui.</li><li>Morbi leo risus, porta ac consectetur ac, vestibulum at eros.</li><li>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</li></ol><figure class=\\"image image--full\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>full image</figcaption></figure><p>Etiam porta sem malesuada magna mollis euismod. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Maecenas sed diam eget risus varius blandit sit amet non magna. Nullam id dolor id nibh ultricies vehicula ut id elit. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><p>Etiam porta sem malesuada magna mollis euismod. Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Maecenas sed diam eget risus varius blandit sit amet non magna. Nullam id dolor id nibh ultricies vehicula ut id elit. Integer posuere erat a ante venenatis dapibus posuere velit aliquet. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><figure class=\\"image image--center image--small\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>centered image, small</figcaption></figure><p>Nullam id dolor id nibh ultricies vehicula ut id elit. Maecenas faucibus mollis interdum. Etiam porta sem malesuada magna mollis euismod. Vestibulum id ligula porta felis euismod semper. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Maecenas sed diam eget risus varius blandit sit amet non magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><p>Nullam id dolor id nibh ultricies vehicula ut id elit. Maecenas faucibus mollis interdum. Etiam porta sem malesuada magna mollis euismod. Vestibulum id ligula porta felis euismod semper. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Maecenas sed diam eget risus varius blandit sit amet non magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><figure class=\\"image image--left image--small\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>left image, small</figcaption></figure><p>Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue.</p><p>Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue.</p><figure class=\\"image image--right image--small\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>right image, small</figcaption></figure><p>Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Donec ullamcorper nulla non metus auctor fringilla. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Maecenas sed diam eget risus varius blandit sit amet non magna. Sed posuere consectetur est at lobortis.</p><p>Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Donec ullamcorper nulla non metus auctor fringilla. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Maecenas sed diam eget risus varius blandit sit amet non magna. Sed posuere consectetur est at lobortis.</p><figure class=\\"image image--center\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>centered image</figcaption></figure><p>Nullam id dolor id nibh ultricies vehicula ut id elit. Maecenas faucibus mollis interdum. Etiam porta sem malesuada magna mollis euismod. Vestibulum id ligula porta felis euismod semper. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Maecenas sed diam eget risus varius blandit sit amet non magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><p>Nullam id dolor id nibh ultricies vehicula ut id elit. Maecenas faucibus mollis interdum. Etiam porta sem malesuada magna mollis euismod. Vestibulum id ligula porta felis euismod semper. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Maecenas sed diam eget risus varius blandit sit amet non magna. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><figure class=\\"image image--left\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>left image</figcaption></figure><p>Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue.</p><p>Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor. Donec ullamcorper nulla non metus auctor fringilla. Nulla vitae elit libero, a pharetra augue.</p><figure class=\\"image image--right\\"><img alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"><figcaption>right image</figcaption></figure><p>Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Donec ullamcorper nulla non metus auctor fringilla. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Maecenas sed diam eget risus varius blandit sit amet non magna. Sed posuere consectetur est at lobortis.</p><p>Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Donec ullamcorper nulla non metus auctor fringilla. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Duis mollis, est non commodo luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit. Maecenas sed diam eget risus varius blandit sit amet non magna. Sed posuere consectetur est at lobortis.</p><img class=\\"image image--full\\" alt=\\"alt\\" title=\\"title\\" srcset=\\"
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=500 500w, 
            /images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=800 8000w
          \\" sizes=\\"100vw\\" src=\\"/images/uploads/headshot_2017.jpg?nf_resize=fit&amp;w=1000\\"></article>"
`;
